python_modules/dagster/dagster/core/asset_defs/asset_group.py

```diff
@@ -639,10 +638,39 @@ def asset2(asset1):
         return AssetGroup(
             assets=result_assets,
             source_assets=self.source_assets,
-            resource_defs={k: r for k, r in self.resource_defs.items() if k != "root_manager"},
+            resource_defs=self.resource_defs,
             executor_def=self.executor_def,
         )
 
+    def __add__(self, other: "AssetGroup") -> "AssetGroup":
+        check.inst_param(other, "other", AssetGroup)
+
+        if self.resource_defs != other.resource_defs:
```

Comment:
why not? I'd expect in most cases they would have different resource_defs. assuming there are no key collisions, any issue with just merging the dictionaries?

Reply:
My concern was that you could end up in situations where a resource in one group unexpectedly applies to assets in another group.  With the upcoming repository-level resources, someone might expect their asset's resource to be coming from the repository and be confused about it sneaking in from another asset group.

Thoughts? We can always loosen this later if we need to.